File: StockTracker/app/src/main/java/com/stocktracker/app/data/model/Stock.kt
package com.stocktracker.app.data.model

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(tableName = "stocks")
data class Stock(
    @PrimaryKey
    val symbol: String,
    val name: String,
    val price: Double,
    val change: Double,
    val changePercent: Double,
    val volume: Long? = null,
    val marketCap: String? = null,
    val peRatio: Double? = null,
    val high52Week: Double? = null,
    val low52Week: Double? = null,
    val lastUpdated: Long = System.currentTimeMillis()
) : Parcelable

@Parcelize
data class StockQuote(
    val symbol: String,
    val open: String,
    val high: String,
    val low: String,
    val price: String,
    val volume: String,
    val latestTradingDay: String,
    val previousClose: String,
    val change: String,
    val changePercent: String
) : Parcelable

@Parcelize
data class SearchResult(
    val symbol: String,
    val name: String,
    val type: String,
    val region: String,
    val marketOpen: String,
    val marketClose: String,
    val timezone: String,
    val currency: String,
    val matchScore: String
) : Parcelable

@Parcelize
data class ChartDataPoint(
    val date: String,
    val price: Double,
    val volume: Long
) : Parcelable

@Entity(tableName = "watchlist")
data class WatchlistItem(
    @PrimaryKey
    val symbol: String,
    val addedAt: Long = System.currentTimeMillis()
)

File: StockTracker/src/types/stock.ts
export interface Stock {
  symbol: string;
  name: string;
  price: number;
  change: number;
  changePercent: number;
  volume?: number;
  marketCap?: string;
  peRatio?: number;
  high52Week?: number;
  low52Week?: number;
}

export interface StockQuote {
  '01. symbol': string;
  '02. open': string;
  '03. high': string;
  '04. low': string;
  '05. price': string;
  '06. volume': string;
  '07. latest trading day': string;
  '08. previous close': string;
  '09. change': string;
  '10. change percent': string;
}

export interface TimeSeriesData {
  [date: string]: {
    '1. open': string;
    '2. high': string;
    '3. low': string;
    '4. close': string;
    '5. volume': string;
  };
}

export interface StockData {
  'Meta Data': {
    '1. Information': string;
    '2. Symbol': string;
    '3. Last Refreshed': string;
    '4. Time Zone': string;
  };
  'Time Series (Daily)': TimeSeriesData;
}

export interface SearchResult {
  '1. symbol': string;
  '2. name': string;
  '3. type': string;
  '4. region': string;
  '5. marketOpen': string;
  '6. marketClose': string;
  '7. timezone': string;
  '8. currency': string;
  '9. matchScore': string;
}

export interface ChartDataPoint {
  date: string;
  price: number;
  volume: number;
}